seq_logger.js

```diff
@@ -1,296 +1,295 @@
 "use strict";
 
-const SafeGlobalBlob = typeof Blob !== 'undefined' ? Blob : require('buffer').Blob;
-const safeGlobalFetch = typeof fetch !== 'undefined' ? fetch : require('node-fetch');
-const SafeGlobalAbortController = typeof AbortController !== 'undefined' ? AbortController : require('abort-controller');
-
-const HEADER = '{"Events":[';
-const FOOTER = "]}";
-const HEADER_FOOTER_BYTES = (new SafeGlobalBlob([HEADER])).size + (new SafeGlobalBlob([FOOTER])).size;
-
-class SeqLogger {
-    constructor(config) {
-        let dflt = {
-            serverUrl: 'http://localhost:5341',
-            apiKey: null,
-            maxBatchingTime: 2000,
-            eventSizeLimit: 256 * 1024,
-            batchSizeLimit: 1024 * 1024,
-            requestTimeout: 30000,
-            maxRetries: 5,
-            retryDelay: 5000,
-            onError: e => { console.error("[seq]", e); }
-        };
-        let cfg = config || dflt;
-        var serverUrl = cfg.serverUrl || dflt.serverUrl;
-        if (!serverUrl.endsWith('/')) {
-            serverUrl += '/';
+module.exports = function (safeGlobalBlob, safeGlobalFetch) {
+    const safeGlobalAbortController = typeof AbortController !== 'undefined' ? AbortController : require('abort-controller');
```

Comment:
We need to have this one injected as well, the main file needs to be 100% require free.

Reply:
Easily done, but I'm not sure it would make a difference. The require would be in both entry points, because Node < 15 needs it and browsers before 2018.

Reply:
If you have this `require`statement in there, the bundler will fail anyhow, so the migration doesn't make sense. Are you sure you need to support browsers from before 2018? Already 6 years ago, even MS completely dropped support for IE.

Alternative is to add `abort-controller` as a dependency to the module, which judging by [npm](https://www.npmjs.com/package/abort-controller), others are doing as well. Then it is 100% there and the build will work.

Reply:
Making the `seq_logger` entirely `require` free sounds good to me, even if some requires are duplicated. It at least makes a clear distinction between it and the entrypoints, and discourages sneaking new ones in.

Reply:
Done. Note that `abort-controller` is already a dependency.